R/zchunk_LB111.ag_resbio_R_C.R

```diff
@@ -32,39 +33,47 @@ module_aglu_LB111.ag_resbio_R_C_DISABLED <- function(command, ...) {
     L100.FAO_ag_Prod_t <- get_data(all_data, "L100.FAO_ag_Prod_t")
     Various_ag_resbio_data <- get_data(all_data, "aglu/Various_ag_resbio_data")
 
-    # ===================================================
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: there are `merge` and/or 'match' calls in this code. Be careful!
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Merge-and-Match
-    # ===================================================
+    # Compute weighted averages of each parameter ( HarvestIndex, ErosionControl, and ResidueEnergyContent ) for each crop type in each GCAM region
+    L100.FAO_ag_Prod_t %>%
+      select(iso, item, year, value) %>%
+      filter(year %in% FINAL_HISTORICAL_YEAR) %>%
+      select(-year) %>%
+      rename(prod = value) %>%
+      full_join(Various_ag_resbio_data, by = "item") %>%
+    # Drop rows with NA values ( dropping commodities that are not in the resbio dataset )
+      na.omit() %>%
+    # also drop rows where production weights are zero, as these would return missing values later on
+      filter(prod != 0) %>%
+    # Multiply by production to get weights, change to long-format for easier calculation
+      gather(resbio_params, value, -c(iso, item, prod)) %>%
+      mutate(value = value * prod) %>%
+    # Add vectors for GCAM regions and commodities, collapse, and divide by production to get residue biomass values
+      left_join(iso_GCAM_regID[c( "iso", "GCAM_region_ID")], by = "iso") %>%
```

Comment:
This should probably be a `left_join_error_no_match`, which throws an error if there's not a 1-to-1 match for everything.